src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -199,6 +199,10 @@ class WorldConfiguration(@PublishedApi internal val world: World) {
     inline fun onRemoveEntity(noinline hook: EntityHook) {
         world.setEntityRemoveHook(hook)
     }
+
+    fun entityProvider(factory: World.() -> EntityProvider) {
+        world.entityService.entityProvider = world.run(factory)
```

Comment:
Need to add a safety here that this gets called first before any other configuration. Because I think otherwise we have a problem when systems are creating entities in their constructor then they will use the Fleks EntityProvider instead of the custom one.

Need to verify that in a test case. Fixing it is easy since we do already something similar with other configuration parts.

Reply:
I think that we can improve this in general by not immediately calling `run` for all configurations like:

```kotlin
fun injectables(cfg: InjectableConfiguration.() -> Unit) = injectableCfg.run(cfg)

fun components(cfg: ComponentConfiguration.() -> Unit) = compCfg.run(cfg)

fun families(cfg: FamilyConfiguration.() -> Unit) = familyCfg.run(cfg)

fun systems(cfg: SystemConfiguration.() -> Unit) = systemCfg.run(cfg)
```

If we delay the run calls, then the user should be able to specify the things in any order he likes and it should still work. We need to adapt `configureWorld` to something like:

```kotlin
val worldCfg = WorldConfiguration(newWorld).apply(cfg)

// NEW STUFF HERE (call configurations in correct order)
worldCfg.injectablesCfg() // = worldCfg.injectablesCfg.run(cfg)
worldCfg.entityProviderFactory()
worldCfg.systemsCfg()
worldCfg.familyCfg()
worldCfg.componentsCfg()

// assign world systems afterward to resize the systems array only once to the correct size
// instead of resizing every time a system gets added to the configuration
newWorld.systems = worldCfg.systems.toTypedArray()
```